common/rpc/middleware.go

```diff
@@ -177,3 +178,15 @@ func (m *HeaderForwardingMiddleware) Call(ctx context.Context, request *transpor
 
 	return out.Call(ctx, request)
 }
+
+// DefaultPartitionConfigMiddleware stores the partition config and isolation group of the request into the context
+type DefaultPartitionConfigMiddleware struct{}
+
+func (m *DefaultPartitionConfigMiddleware) Handle(ctx context.Context, req *transport.Request, resw transport.ResponseWriter, h transport.UnaryHandler) error {
```

Comment:
Should this have a guard on config to determine if the isolation-group-feature is enabled? elsewhere - unless I'm confused, it looked like you were using the presence of the poller Isolation-group to determine if the feature is enabled.

Reply:
this is NOT the most suitable place to do the check. 
1. We only get the request which is not deserialized and are not able to get the domain name.
2. It's hard to introduce a dependency to middleware because cadence is not an Fx application.

Reply:
The best place is `workflowHandler.go`. We can add a check, but I don't think it's necessary. 
It can only save a few bytes storage per workflow for domains not having this feature enabled.

Reply:
yeah, I think I wrote this message before we talked and I was just confused, what you have seems reasonable